src/pybind/mgr/prometheus/module.py

```diff
@@ -177,7 +177,7 @@ class HealthHistory:
 
     def __init__(self, mgr: MgrModule):
         self.mgr = mgr
-        self.lock = threading.Lock()
+        self.lock = threading.RLock()
```

Comment:
The other users of the `lock` property are:

```python
    def reset(self) -> None:
        """Reset the healthcheck history."""
        with self.lock:
            self.mgr.set_store(self.kv_name, "{}")
            self.healthcheck = {}

    def save(self) -> None:
        """Save the current in-memory healthcheck history to the KV store."""
        with self.lock:
            self.mgr.set_store(self.kv_name, self.as_json())
```

Please note that `RLock` in Python is **not** the _read-write lock_, it's the [_reentrant lock_](https://docs.python.org/3/library/threading.html#rlock-objects).